src/behaviors/Behavior.js

```diff
@@ -1113,39 +1120,34 @@ var Behavior = Base.extend('Behavior', {
 
     /**
      * @memberOf Behavior.prototype
-     * @return {Object} The properties for a specific column. These are used if no cell properties are specified.
+     * @return {Object} The properties for a specific column.
      * @param {index} columnIndex - the column index of interest
      */
-    getColumnProperties: function(columnIndex) {
-        var column = this.columns[columnIndex];
-        if (!column) {
-            return isNull;
-        }
-        var properties = column.getProperties(); //TODO: returns `null` on Hypergrid.prototype.reset();
-        if (!properties) {
-            return isNull;
-        }
-        return properties;
+    getColumnProperties: function(x) {
+        var column = this.columns[x];
+        return column && column.getProperties();
     },
 
-    setColumnProperties: function(columnIndex, properties) {
-        var column = this.allColumns[columnIndex];
-        var columnProperties = column.getProperties();
-        _(columnProperties).extendOwn(properties);
+    setColumnProperties: function(x, properties) {
+        var column = this.getColumn(x);
+        if (!column) {
+            throw 'Expected data column.';
```

Comment:
Maybe remove the word data as you can set properties for headers inside of the columns as well